central/complianceoperator/v2/report/manager/watcher/scanconfigwatcher.go

```diff
@@ -117,6 +122,7 @@ func (w *scanConfigWatcherImpl) run(timerC <-chan time.Time) {
 	defer func() {
 		w.stopped.Signal()
 		<-w.stopped.Done()
+		w.timeout.Stop()
```

Comment:
Note, that now this func accepts `timerC` and manipulates the `timer` field. They both come from the same object! I think this could be refactored.

Reply:
The `timerC` is only passed for testing purposes. To be honest I don't like it the fact that I use the `timerC` in the loop but then I call the `timeout` field. Let me think about it and see if I can come up with a nice interface for this. Otherwise I can change the tests to have a very small timer and use `Eventually` functions.

Reply:
So, I went if a `stopFn` to not have the `timeout` field. To be honest I don't really like this solution either but I would really try to avoid as much as possible losing the ability to inject a channel to run the function as losing it will make the tests more unpredictable.

Reply:
I agree it is not perfect, but I also think it is important to have the channel as a parameter.

I was thinking about something like this:  
```go
type stoppable interface {
	Stop() bool
}

func (w *scanConfigWatcherImpl) run(timerC <-chan time.Time, timerStop stoppable) {...}
```

and then call it with

```go
go ret.run(timeout.C, timeout)
```

but that is also not perfect...

I think the current state is good enough, so I will not block the entire PR for this. We can think about options for stopping the timer later.

Reply:
You're going to laugh but I thought of doing something like that too 😆, I ended up not doing it because you could technically do the same thing here, pass a channel and a different timeout (which you can also do with the `stopFn`. I think the only solution to not have this problem is to do a wrapper of the `Timer`, something like:

```golang
type timer interface {
    Stop() bool
    C() <-chan time.Time
}

type timerImpl struct {
    timeout *time.Timer
}
func (t *timerImpl) C() <-chan time.Timer {
    return t.timeout.C
}
func (t *timerImpl) Stop() bool {
    return t.timeout.Stop()
}
```

and then pass this as:

```golang
go ret.run(timeoutWrapper)
```